microcosm_logging/tests/test_decorators.py

```diff
@@ -15,7 +19,43 @@ class TestClass(object):
     pass
 
 
+@logger
+class TestContextClass(object):
+
+    def function(self):
+        self.logger.info("success!")
+
+
 def test_using_class_logger_works():
     instance = TestClass()
 
     assert_that(instance.logger, is_(equal_to(getLogger(instance.__class__.__name__))))
+
+
+def test_using_context_logger_works():
+    instance = TestContextClass()
+    context_func = Mock(return_value={'some': 'context'})
+    wrapped = context_logger(context_func, instance.function, instance)
```

Comment:
How about writing a test case that looks more like the actual usage, e.g. writing a class and decorating it?

I'm sort of vague on how this usage would actually work. We'd decorate the class and then re-decorate specific functions? I think concrete usage example would help.